src/NanoMsg.h

```diff
@@ -0,0 +1,19 @@
+#include <nanomsg/nn.h>
+#include <string>
+#include <vector>
+/*
+ * class for encapsulating nanomsg zero copy allocation/deletion
+ */
+class NanoMsg {
+   typedef const std::vector<std::pair<void*, unsigned int>> PacketHashVector;
+public:
+   NanoMsg(PacketHashVector& data);
+   NanoMsg(void*& data);
+   NanoMsg(const std::string& data);
+   ~NanoMsg();
+   NanoMsg& operator=(const NanoMsg&) = delete;
+   NanoMsg(const NanoMsg&) = delete;
+   NanoMsg() = delete;
+   void* buffer;
```

Comment:
we can if we really wanted to. i just found it to be kind of superfluous. the buffer is only ever accessed on the sender and i thought it was just extra work to add an accessor. the class itself should not be used outside of Queuenado, maybe we can make the entire class private?